### DISCUSSION TITLE
[Feature Request]: Combine coilgun ammo boxes to top them up.

### DISCUSSION THREAD

**Discussion by RaptorIM** [FakeFishGames/Barotrauma#12352] [Ideas and suggestions]
- [X] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Allow used coilgun ammo boxes to be combined together to transfer capacity from one to the other. When one is emptied by this completely, empty box remains behind.

**Additional information**
Same mechanic as with SMG magazines that can be to topped-off from other SMG magazine, allow this for coilgun ammo boxes.

**Comment by Regalis11** [comment]
The problem with combining coilgun ammo boxes is that you can only have one box in your inventory, so you can't drag them to the same slot the same way as SMG magazines or other combinable items. The only way to combine the boxes would be to drag them to the same loader (which currently swaps them) or to combine them in an ammo shelf, and I think making the former combine the boxes instead of swapping them might lead to some confusion.

@ezjamsen @danielasteljoki Any thoughts on this?

**Reply by Hazmat616** [reply]
I made a mod called one handed ammo boxes that does this as well as its namesake. While holding two does allow them to be combined like smgs. The ability to combine them on the shelf or loader is very intuitive and no one has mentioned it being confusing.

**Reply by mygamingaccount** [reply]
A very popular mod exists on the Workshop that lets you combine ammo. There are no problems, it's a direct improvement.
https://steamcommunity.com/sharedfiles/filedetails/?id=2799051561

**Comment by RaptorIM** [comment]
Would it be possible to check if the item in container is:
different type than the one in hands -> swap;
same type and full -> swap;
same type, container in hands is empty so nothing can be transferred -> swap;
same type and not full -> transfer content from item in hands to item in container?
This would allow to simply fill up non-full containers if we have the same one, and worst case scenario if there is a full one already and we exchange it for empty, we can refill the empty one with a single click.

**Comment by EtiamNullam** [comment]
I'm strongly against this feature - If they were combinable people would just reload them at every occasion. It will make it too easy to keep coilguns loaded all the time without funny dilemma: "Should I switch or will it last through next enemy encounter" or interaction that requires cooperation: "You shoot, I will reload once empty".

Combinable coilgun ammo is a step backward.

**Comment by RaptorIM** [comment]
And how is that different to swapping for full, deconstructing the half-empty ones, and merging the materials, apart from this being a boring pain to do? Simple quality of life feature.

**Comment by thegrb93** [comment]
Because doing that costs time and energy resource. A valid cost imo. Part of the game is resource management, boring to you maybe but not to others.

**Comment by EtiamNullam** [comment]
Great explanation from @thegrb93 but I just wanted to add something @RaptorIM apparently forgot: you can simply shoot the boxes empty instead of deconstructing. Do reload with a full one before it runs out when you expect difficult encounter soon but leave box with 30% ammo left lying there as its more useful than 30% ingot of steel (or other ingredient) and could come in handy if you run low, and saves you "boring pain" of reconstructing.

**Comment by G-M-twostay** [comment]
maybe combine them in fabricator? solves the problem of being able to only hold one box in hand.

**Comment by VaelophisNyx** [comment]
This feature exists in a mod and is a direct improvement to gameplay. People need to realize tedium is not fun in the least

**Comment by Ek-Videogames** [comment]
To the morons complaining about 'realism': We **_had_** a 'realistic' way to combine partial coilgun boxes by crafting, but that was by letting deconstructed ammo produce partially-used materials _(which you could then combine the materials back to full)_ which got removed for the sake of simplicity.

Also I don't know about you, but it seems pretty simple to just link two partial belts of ammo together to make a new single belt.

**Reply by mygamingaccount** [reply]
And what an absolute pain in the ass that was! Having partial _anything_ was nothing but a nuisance. So glad that it's no longer a thing.
Now instead of having 5% steel or whatever, that you would normally flush down the airlock to get rid of it, Barotrauma automatically deletes it for you.

**Reply by mygamingaccount** [reply]
The folks "complaining about realism" are inside your head. Not a single comment is about realism.

**Reply by Ek-Videogames** [reply]
Leave it to the clown to be the only person arguing that they don't want ammo boxes to be combinable. 
Go punch an endworm and stop denying every single QOL addition because it will make the game "not cbt enough!"

**Reply by EtiamNullam** [reply]
> Leave it to the clown to be the only person arguing that they don't want ammo boxes to be combinable.

Don't pretend it's only a single person if 5 people downvoted the idea that it's a "Simple quality of life feature."
